odoo/addons/base/models/ir_model.py

```diff
@@ -2210,8 +2210,16 @@ def _module_data_uninstall(self, modules_to_remove):
             model = self.pool.get(ir_field.model)
             if model is not None:
                 field = model._fields.get(ir_field.name)
-                if field is not None:
-                    field.prefetch = False
+                if field is not None and field.prefetch:
+                    if not field._toplevel:
+                        # avoid modifying non-_toplevel field objects since
+                        # they may be shared with other registries
+                        Field = type(field)
+                        field_ = Field(_base_fields=[field, Field(prefetch=False)])
+                        self.env[ir_field.model]._add_field(ir_field.name, field_)
+                        field_.setup(model)
+                    else:
+                        field.prefetch = False
```

Comment:
I agree with your analysis.  Indeed, strategy 1 looks like the right thing to do in stable branches.
Simple code suggestion: make the `if` statement more direct.
```py
if field._toplevel:
    # the field is specific to this registry
    field.prefetch = False
else:
    # the field is shared across registries; don't modify it
    ...
```